Modules/Microsoft365DSC/DSCResources/MSFT_AzureBillingAccountPolicy/MSFT_AzureBillingAccountPolicy.psm1

```diff
@@ -0,0 +1,475 @@
+            -Exception $_ `
+            -Source $($MyInvocation.MyCommand.Source) `
+            -TenantId $TenantId `
+            -Credential $Credential
+
+        return $nullResult
+    }
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $BillingAccount,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance]
+        $EnterpriseAgreementPolicies,
+
+        [Parameter()]
+        [System.String]
+        $MarketplacePurchases,
+
+        [Parameter()]
+        [System.String]
+        $ReservationPurchases,
+
+        [Parameter()]
+        [System.String]
+        $SavingsPlanPurchases,
+
+        [Parameter()]
+        [System.String]
+        $Name,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint,
+
+        [Parameter()]
+        [Switch]
+        $ManagedIdentity,
+
+        [Parameter()]
+        [System.String[]]
+        $AccessTokens
+    )
+
+    New-M365DSCConnection -Workload 'Azure' `
+        -InboundParameters $PSBoundParameters | Out-Null
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName.Replace('MSFT_', '')
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $instanceParams = @{
+        properties = @{
+            enterpriseAgreementPolicies = @{
+                accountOwnerViewCharges    = $EnterpriseAgreementPolicies.accountOwnerViewCharges
+                authenticationType         = $EnterpriseAgreementPolicies.authenticationType
+                departmentAdminViewCharges = $EnterpriseAgreementPolicies.departmentAdminViewCharges
+            }
+            marketplacePurchases = $MarketplacePurchases
+            reservationPurchases = $ReservationPurchases
+            savingsPlanPurchases = $SavingsPlanPurchases
+        }
+    }
+    $payload = ConvertTo-Json $instanceParams -Depth 5 -Compress
+    Write-Verbose -Message "Updating billing account policy for {$BillingAccount} with payload:`r`n$($payload)"
+    $uri = "https://management.azure.com/providers/Microsoft.Billing/billingAccounts/$($BillingAccount)/policies/default?api-version=2024-04-01"
+    $response = Invoke-AzRest -Uri $uri -Method "PUT" -Payload $payload
```

Comment:
What happens when this request fails? For example because of missing permissions. The code does not handle this currently, so will crash. Maybe good to add some error handling logic.